Sources/Datadog/DatadogCore/DatadogCore.swift

```diff
@@ -182,6 +186,24 @@ internal final class DatadogCore {
             self.messageBusQueue.async { messageReceiver.receive(message: .context(context), from: self) }
         }
     }
+
+    /// A list of storage units of currently registered Features.
+    private var allStorages: [FeatureStorage] {
+        let v1Storages = v1Features.values.compactMap { $0 as? V1Feature }.map { $0.storage }
+        let v2Storages = v2Features.values.map { $0.storage }
+        return v1Storages + v2Storages
```

Comment:
I'm curious if we could end up in a state where we could have same feature in both v1 and v2 (middle of migration maybe?).

If yes - maybe it makes sense to perform union instead of adding the two? 🤔

Reply:
> I'm curious if we could end up in a state where we could have same feature in both v1 and v2 (middle of migration maybe?).

Very unlikely - we have only 3 features registered in V1 and quite clear migration path for each. I made this code verbose to facilitate the ultimate change we will do after migrating the last one:
```diff
-        let v1Storages = v1Features.values.compactMap { $0 as? V1Feature }.map { $0.storage }
-        let v2Storages = v2Features.values.map { $0.storage }
-        return v1Storages + v2Storages
+ return features.values.map { $0.storage }
```